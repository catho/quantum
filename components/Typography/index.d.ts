import { FC } from 'react';

declare const Typography: FC;

export default Typography;
