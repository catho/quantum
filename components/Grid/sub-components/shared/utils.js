import { hideQueries } from './media';

const hide = ({ hide: hideProp, theme: { breakpoints } }) =>
  hideProp &&
  []
    .concat([], hideProp)
    .map(breakpoint => hideQueries(breakpoints)[breakpoint]());

const calcGutter = (gutter, noGuttersProp, onlyValue = false) => {
  let returnValue = '0';
  if (gutter) {
    returnValue = noGuttersProp ? '0' : `${gutter}`;
  }

  return `${returnValue.replace('px', '')}${onlyValue ? '' : 'px'}`;
};

export { hide, calcGutter };
