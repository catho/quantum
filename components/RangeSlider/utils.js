const mountMarksByArray = (values, labelFunction) =>
  values.map(value => ({
    value,
    label: labelFunction ? labelFunction(value) : value,
  }));

export default {
  mountMarksByArray,
};
