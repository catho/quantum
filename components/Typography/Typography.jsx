const Typography = () => (
  <>
    <link
      rel="preconnect"
      href="https://fonts.gstatic.com"
      crossOrigin="anonymous"
    />
    <link
      rel="preload"
      href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
      as="font"
      crossOrigin="anonymous"
    />
    <link
      href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
      rel="stylesheet"
    />
  </>
);

export default Typography;
