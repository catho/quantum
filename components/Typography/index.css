@import "./sub-components/Blockquote.module.css";
@import "./sub-components/Subtitle.module.css";