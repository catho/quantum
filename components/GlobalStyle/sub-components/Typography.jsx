import React from 'react';
import { createGlobalStyle } from 'styled-components';
import { colors } from '../../shared/theme';
import isSSR from '../../shared/isSSR';

const Style = createGlobalStyle`
* {
  font-family: 'Nunito Sans', sans-serif;
  line-height: 1.5;
}

body {
  font-size: 16px;
  letter-spacing: 0.2px;
  color: ${colors.neutral['700']};
}

svg[data-qtm-preloader="icon"] {
  max-width: 0;
}

h1, h2, h3, h4, h5, h6 {
  margin: 0;
  font-weight: normal;
}

h1 {
  letter-spacing: 0.7px;
  font-size: 48px;
}

h2 {
  font-size: 36px;
  letter-spacing: 0.5px;
}

h3 {
  font-size: 28px;
  letter-spacing: 0.4px;
}

h4 {
  font-size: 24px;
  letter-spacing: 0.4px;
}

h5 {
  font-size: 20px;
  letter-spacing: 0.3px;
}

h6 {
  font-size: 18px;
  letter-spacing: 0.3px;
}

ul, ol {
  margin: 0;
  padding-left: 18px;
}`;

const NunitoFontFamily = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap');
`;

const Typography = () => (
  <>
    <Style />
    {isSSR() ? (
      <link
        href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
        rel="stylesheet"
      />
    ) : (
      <NunitoFontFamily />
    )}
  </>
);

export default Typography;
