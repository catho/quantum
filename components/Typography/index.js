import Typography from './Typography';
import Title from './sub-components/Title';
import Subtitle from './sub-components/Subtitle';
import Blockquote from './sub-components/Blockquote';

export { Title, Subtitle, Blockquote };
export default Typography;
