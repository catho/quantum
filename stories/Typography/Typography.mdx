import { Meta, Story } from '@storybook/addon-docs';

import { Header } from '../shared'
import { Headings, Texts, Lists } from './Components'
import EvaluationForm from '../../stories/shared/EvaluationForm'


<Meta title="Foundation/Typography" />

<Header title="Typography">
  Typography - Quantum uses Nunito Sans as the standard typeface to present
  our design and content as clearly and efficiently as possible.
</Header>

- [Importing and usage](#importing)
- [Examples](#examples)

## <a id="importing"></a>Importing and Usage


```js
import Typography from '@catho/quantum/Typography';
```

Usually, it is used before the application component

```js
ReactDOM
  .render(() => (
    <>
      <Typography />
      <App />
    </>
  ), document.getElementById('root'));
```

The `<Typography />` will add and set automatically all the base configuration for font and [Icons](?path=/docs/foundation-icons--page).

If you want to include one of those separately, you can import each as you'd like to:

```js
import Typography from '@catho/quantum/Typography';
```
## <a id="examples"></a>Examples
## Components

### Headings

<Headings />

### Texts

<Texts />

### Lists

more info about List component, click [here](?path=/docs/foundation--list-story).

```js
import { List } from '@catho/quantum';
```

```js
const exampleItemsDefault = [
  { content: 'Lorem ipsum dolor sit amet' },
  { content: 'Sed do eiusmod tempor ' },
  { content: 'Consectetur adipiscing elit' },
];
```

<Lists />

<EvaluationForm componentName="Typography" />
