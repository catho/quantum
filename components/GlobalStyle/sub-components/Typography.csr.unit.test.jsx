import React from 'react';
import renderer from 'react-test-renderer';
import Typography from './Typography';

import { expectCSSMatches } from '../test.utils';

describe('Typography component', () => {
  it('Should match snapshot', () => {
    const typography = renderer.create(<Typography />);

    expect(typography).toMatchSnapshot();
  });

  it('Should set standard typeface', () => {
    renderer.create(<Typography />);

    expectCSSMatches(`* {
      font-family: 'Nunito Sans', sans-serif;
      line-height: 1.5;
    }`);
  });

  it('Should set h1 style', () => {
    renderer.create(<Typography />);

    expectCSSMatches(`h1 {
        -webkit-letter-spacing: 0.7px;
        -moz-letter-spacing: 0.7px;
        -ms-letter-spacing: 0.7px;
        letter-spacing: 0.7px;
        font-size: 48px;
      }`);
  });

  it('Should set h2 style', () => {
    renderer.create(<Typography />);

    expectCSSMatches(
      `h2 {
        font-size: 36px;
        -webkit-letter-spacing: 0.5px;
        -moz-letter-spacing: 0.5px;
        -ms-letter-spacing: 0.5px;
        letter-spacing: 0.5px;
      }`,
    );
  });

  it('Should set h3 style', () => {
    renderer.create(<Typography />);

    expectCSSMatches(
      `h3 {
        font-size: 28px;
        -webkit-letter-spacing: 0.4px;
        -moz-letter-spacing: 0.4px;
        -ms-letter-spacing: 0.4px;
        letter-spacing: 0.4px;
      }`,
    );
  });

  it('Should set h4 style', () => {
    renderer.create(<Typography />);

    expectCSSMatches(
      `h4 {
        font-size: 24px;
        -webkit-letter-spacing: 0.4px;
        -moz-letter-spacing: 0.4px;
        -ms-letter-spacing: 0.4px;
        letter-spacing: 0.4px;
      }`,
    );
  });

  it('Should set h5 style', () => {
    renderer.create(<Typography />);

    expectCSSMatches(
      `h5 {
        font-size: 20px;
        -webkit-letter-spacing: 0.3px;
        -moz-letter-spacing: 0.3px;
        -ms-letter-spacing: 0.3px;
        letter-spacing: 0.3px;
      }`,
    );
  });

  it('Should set h6 style', () => {
    renderer.create(<Typography />);

    expectCSSMatches(
      `h6 {
        font-size: 18px;
        -webkit-letter-spacing: 0.3px;
        -moz-letter-spacing: 0.3px;
        -ms-letter-spacing: 0.3px;
        letter-spacing: 0.3px;
      }`,
    );
  });
});
