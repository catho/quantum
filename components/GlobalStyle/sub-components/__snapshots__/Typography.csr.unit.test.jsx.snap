// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component Should match snapshot 1`] = `
<link
  href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,600,700,400i,600i,700i&display=swap"
  rel="stylesheet"
/>
`;
